src/utils/inject-stylesheet.js

```diff
@@ -7,13 +11,56 @@
  *
  * @returns {void}
  */
-export function injectStylesheet(endpoint, env, version, uid, cid) {
-	const queryParams = (cid) ? '?previewcid=' + cid : '';
+export function injectStylesheetWithAutoReplace(endpoint, env, version, uid, cid, attempt = 0) {
+	injectStylesheet(endpoint, env, version, uid, cid)
+	const url = getStyleSheetUrl(endpoint, env, version, uid, cid);
+	const cssFile = getCSSFile(url);
+
+	if (cssFile && window.MutationObserver) {
+		const observer = new window.MutationObserver(function (mutationList) {
+			mutationList.forEach(function (mutation) {
+				mutation.removedNodes.forEach(function (node) {
```

Comment:
This can potentially result in the auto replace being called multiple times per attempt no? Should we change these `forEach` calls into something like an if `(mutationList.some(mutation => mutation.removedNodes.some)))`?

Reply:
this is directly javascript code and we're not transpiling. So I'll avoid some. But I will return if we re-add the css